lib/ecto_diff.ex

```diff
@@ -276,10 +300,12 @@ defmodule EctoDiff do
     }
   end
 
-  defp fields(%struct{} = previous, %struct{} = current) do
-    field_names = struct.__schema__(:fields) ++ (struct.__schema__(:virtual_fields) -- struct.__schema__(:embeds))
+  defp fields(%struct{} = previous, %struct{} = current, opts) do
+    all_fields = struct.__schema__(:fields) ++ (struct.__schema__(:virtual_fields) -- struct.__schema__(:embeds))
 
-    field_names
+    selected_fields = if_only_select_fields(all_fields, current, opts)
```

Comment:
:warning: Due to the recursive nature of EctoDiff, we'll be selecting these specific fields regardless of the nesting. So a `:pet` that is under `[:foo]` and `:pet` under `[:bar, :baz]` will both be filtered. What if the user only wants to filter the first pet but not the second?

Not a big fan of this PR, tbh. The point of EctoDiff is to show what changed between two structs. If the caller wants to ignore certain changes, they should be able to it after the diff has already been calculated.

We could expose some helper functions that operate on an existing EctoDiff in order to do that. For example, we can define a `def drop_changes/1` function which takes an EctoDiff and drops any changes to non-association and non-embedded fields. We can also a `drop_assoc_changes/3` and `drop_embed_changes/3` which take the EctoDiff, the name of the field, and a function that takes an EctoDiff and returns and EctoDiff. This would give us a nice API for recursively dropping changes from EctoDiffs in a more granular manner.

Reply:
This is the reason that this is a non-default behavior that has to be explicitly selected, and configured, and that EctoDiff does not change.
If you for some reason have the data structure:
```
%{
  bar: %{pet: %{...}},
  baz: %{bar: %{pet: %{...}}}
}
```
There is no way to only select one of the `bar.pet` instances. You can select both, everything, or none. 
However if this is not the case, then you gain the ability of not selecting to diff a pet that you don't want so long as the parent's are different.